app-server/routes/security.js

```diff
@@ -48,12 +37,101 @@ router.route(`/:symbol`)
+      const IndexPriceData = await PricedataCollection.findOne({"metadata.symbol":"SPY"}, { projection: { data: 1} });
+      let dateCheck;
+      let tickerShares,indexShares;
+
+      if(process.env.NODE_ENV === `production`){
+        dateCheck = new Date();
+        while(!businessDays(dateCheck.toLocaleDateString(),'MM/DD/YYYY').isBusinessDay()) {
+          dateCheck.setDate( dateCheck.getDate() - 1);
+          //console.log("Date Check: " + dateCheck.toLocaleDateString())
+        }
+        dateCheck.setMonth(dateCheck.getMonth()-6);
+        dateCheckConverted = await dateConversion (dateCheck);
+        tickerShares= 100000/tickerPriceData.data[dateCheckConverted].adjustedClose;
+        indexShares= 100000/IndexPriceData.data[dateCheckConverted].adjustedClose;
+      }
+      else {
+        dateCheck = new Date("2019-03-05");
+        dateCheck.setMonth(dateCheck.getMonth()-6);
+        dateCheckConverted = await dateConversion (dateCheck);        
+        tickerShares = 100000/tickerPriceData.data[dateCheckConverted].adjustedClose;
+        indexShares = 100000/IndexPriceData.data[dateCheckConverted].adjustedClose;
+
+      }
+
+      let historicalData = [];
+      const indexDataValues = Object.values(IndexPriceData.data);
+    
+      const symbolDataKeys = Object.keys(tickerPriceData.data);
+      const symbolDataValues = Object.values(tickerPriceData.data);
+    
+      const portfolioTickerName = "Portfolio"+symbol;
+      console.log(dateCheck.toLocaleDateString())
+      for (let index = 0; ; index += 5) {
+        if (new Date(symbolDataKeys[index]) <= dateCheck) {
+          historicalData.push({
+            date: symbolDataKeys[index],
+            SPY: 0,
+            [symbol]: 0,
+            [portfolioTickerName]: (tickerShares * symbolDataValues[index].adjustedClose).toFixed(0),
+            PortfolioSPY: (indexShares * indexDataValues[index].adjustedClose).toFixed(0)
+          });
+          break;
+        }
+        else {
+          const symbolNewPrice = symbolDataValues[index].adjustedClose;
+          const symbolOldPrice = symbolDataValues[index + 1].adjustedClose;
+      
+          const indexNewPrice = indexDataValues[index].adjustedClose;
+          const indexOldPrice = indexDataValues[index + 1].adjustedClose;
+          historicalData.push({
+            date: symbolDataKeys[index],
+            SPY: (indexNewPrice - indexOldPrice) / indexOldPrice,
+            [symbol]: (symbolNewPrice - symbolOldPrice) / symbolOldPrice,
+            [portfolioTickerName]: (tickerShares * symbolDataValues[index].adjustedClose).toFixed(0),
+            PortfolioSPY: (indexShares * indexDataValues[index].adjustedClose).toFixed(0)
+          });
+        }
+      }
+      historicalData = historicalData.reverse();
+      return response.status(200).send(historicalData);
+    }
+    catch (error) {
+      return next(error);
+    }
+  });
+
+ async function dateConversion(input) {
```

Comment:
This function can be made more clear. I get what it does and while it is succinct, it's not obvious what it's doing.

Reply:
I also don't believe that it has to be `async`